src/docfx.website.themes/common/common.js

```diff
@@ -103,36 +102,64 @@ function normalizeGitUrlToHttps(repo) {
     return 'https://' + repo.substr(pos + 1).replace(/:/g, '/');
 }
 
-function getNewFileUrl(uid, gitContribute, gitUrlPattern) {
+function getNewFileUrl(item, gitContribute, gitUrlPattern) {
     // do not support VSO for now
-    if (!gitContribute || !gitContribute.repo) return '';
-    var repo = gitContribute.repo;
-    if (repo.substr(-4) === '.git') {
-        repo = repo.substr(0, repo.length - 4);
+    if (!item.source) {
+        return '';
     }
 
-    var patternName = getPatternName(repo, gitUrlPattern);
+    var repoInfo = getGitContributeFallbackWithGitRemote(item.source.remote, gitContribute);
+    if (!repoInfo.repo || !repoInfo.branch || !repoInfo.path) {
+        return '';
+    }
+
+    if (repoInfo.repo.substr(-4) === '.git') {
+        repoInfo.repo = repoInfo.repo.substr(0, repoInfo.repo.length - 4);
+    }
+
+    var patternName = getPatternName(repoInfo.repo, gitUrlPattern);
     if (!patternName) return patternName;
-    return gitUrlPatternItems[patternName].generateNewFileUrl(repo, gitContribute.branch, gitContribute.path, uid);
+    return gitUrlPatternItems[patternName].generateNewFileUrl(repoInfo, item.uid);
 }
 
 function getRemoteUrl(remote, startLine, gitContribute, gitUrlPattern) {
-    var repo = undefined;
-    var branch = undefined;
-    if (gitContribute && gitContribute.repo) repo = gitContribute.repo;
-    if (repo == undefined && remote && remote.repo) repo = remote.repo;
-    if (gitContribute && gitContribute.branch) branch = gitContribute.branch;
-    if (branch == undefined && remote && remote.branch) branch = remote.branch;
-
-    if (repo == undefined || branch == undefined) return '';
+    var repoInfo = getGitContributeFallbackWithGitRemote(remote, gitContribute);
+    if (!repoInfo.repo || !repoInfo.branch || !repoInfo.path) {
+        return '';
+    }
 
-    if (repo.substr(-4) === '.git') {
-        repo = repo.substr(0, repo.length - 4);
+    if (repoInfo.repo.substr(-4) === '.git') {
+        repoInfo.repo = repoInfo.repo.substr(0, repoInfo.repo.length - 4);
     }
 
-    var patternName = getPatternName(repo, gitUrlPattern);
+    var patternName = getPatternName(repoInfo.repo, gitUrlPattern);
     if (!patternName) return '';
-    return gitUrlPatternItems[patternName].generateUrl(repo, branch, remote.path, startLine);
+
+    repoInfo.startLine = startLine;
+    return gitUrlPatternItems[patternName].generateUrl(repoInfo);
+}
+
+function getGitContributeFallbackWithGitRemote(gitRemote, gitContribute) {
+    var repoInfo = {};
+    if (gitContribute && gitContribute.repo) {
+        repoInfo.repo = gitContribute.repo;
+    } else if (gitRemote && gitRemote.repo) {
+        repoInfo.repo = gitRemote.repo;
+    }
+
+    if (gitContribute && gitContribute.branch) {
+        repoInfo.branch = gitContribute.branch;
+    } else if (gitRemote && gitRemote.branch) {
+        repoInfo.branch = gitRemote.branch;
+    }
+
+    if (gitContribute && gitContribute.path) {
```

Comment:
>path [](start = 39, length = 4)

seems path has different meaning in "source" and "gitContribute", use another property?